backends/intel_gpu/kernels/argsort_kernel.cc

```diff
@@ -0,0 +1,221 @@
+
+    phi::DenseTensor tmp_out;
+    tmp_out.Resize(trans_dims);
+    T* t_out = dev_ctx.template Alloc<T>(&tmp_out);
+
+    phi::DenseTensor tmp_indices;
+    tmp_indices.Resize(trans_dims);
+    auto* t_ind = dev_ctx.template Alloc<int64_t>(&tmp_indices);
+
+    FullSort<T, int64_t>(input_height,
+                         input_width,
+                         in_dims.size(),
+                         &trans_inp,
+                         t_out,
+                         t_ind,
+                         descending);
+
+    dev_ctx.template Alloc<int64_t>(indices);
+    TransposeKernel<int64_t>(dev_ctx, tmp_indices, trans, indices);
+    // transpose back
+    TransposeKernel<T>(dev_ctx, tmp_out, trans, output);
+  }
+}
+
+
+
+
+namespace gpu {
+
+template <typename T>
+void ArgsortKernel(const phi::Context& dev_ctx,
+                   const phi::DenseTensor& input,
+                   int axis,
+                   bool descending,
+                   phi::DenseTensor* output,
+                   phi::DenseTensor* indices) {
+  auto in_dims = input.dims();
+  axis = (axis < 0) ? (in_dims.size() + axis) : axis;
+  T* out_data = dev_ctx.template Alloc<T>(output);
+
+  int64_t* ids_data = dev_ctx.template Alloc<int64_t>(indices);
+  show_kernel("argsort in_dims=" << in_dims << " axis="<< axis << " type="<< dnn_support::type2String<T>::name() << " desc="<< descending );
+
+  PD_CHECK(in_dims.size()<3, "PoC Lenet/Mnist use case only");
+
+  using namespace oneapi::dpl::execution;
+  using namespace oneapi::dpl;
+  auto* q = static_cast<sycl::queue*>(const_cast<void*>(dev_ctx.stream()));
+  auto policy_e = make_device_policy(*q);
+
+  size_t n = 1;
+  size_t m = in_dims[0];
+
+  if(in_dims.size()==2)
+  {
+     n = in_dims[0];
+     m = in_dims[1];
+  }
+
+  auto input_data = input.data<T>();
+  q->memcpy(out_data,input_data, input.memory_size() );
+  q->wait();
+
+  for(size_t i=0;i<n;i++)
+  {
+
+   q->parallel_for(m, [p_data=ids_data + i*m,m](auto& i){
+           p_data[i] = i;
+   });
+
+   q->wait();
+
+   sycl::buffer<int64_t> keys_buf{reinterpret_cast<int64_t*>(ids_data + i*m),sycl::range<1>(m)};
+   sycl::buffer<T> vals_buf{reinterpret_cast<T*>(out_data + i*m),sycl::range<1>(m)};
+
+   auto keys_begin = oneapi::dpl::begin(keys_buf);
+   auto vals_begin = oneapi::dpl::begin(vals_buf);
+   auto zipped_begin = dpl::make_zip_iterator(keys_begin, vals_begin);
+
+    // gpu sort
+     std::stable_sort(policy_e, zipped_begin, zipped_begin + m,[descending](auto lhs, auto rhs) {
+        return (descending)? (get<1>(lhs) > get<1>(rhs)) :  (get<1>(lhs) < get<1>(rhs));
+     });
+
+   }
+
+} // ArgsortKernel
+
+} // gpu
+
+}  // namespace custom_kernel
+
+PD_BUILD_PHI_KERNEL(argsort,
+                    custom_cpu,
+                    ALL_LAYOUT,
+                    custom_kernel::ArgsortKernel,
+                    float,
+                    double,
+                    int,
+                    int64_t) {}
```

Comment:
Please delete redundant custom cpu code, only intel_gpu backend kernel needs to be registered.

Reply:
removed